tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3465,3 +3465,362 @@ def create_zeros_like_3_val():
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Tuples not implemented yet"),
+            pytest.mark.c]
+        )
+    )
+)
+
+def test_numpy_real_array_like_1d(language):
+
+    def test_bool():
+        from numpy import real, shape, array, bool
+        arr = array([4,5,6,2,1], bool)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_int():
+        from numpy import real, shape, array, int
+        arr = array([4,5,6,2,1], int)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_int8():
+        from numpy import real, shape, array, int8
+        arr = array([4,5,6,2,1], int8)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_int16():
+        from numpy import real, shape, array, int16
+        arr = array([4,5,6,2,1], int16)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_int32():
+        from numpy import real, shape, array, int32
+        arr = array([4,5,6,2,1], int32)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_int64():
+        from numpy import real, shape, array, int64
+        arr = array([4,5,6,2,1], int64)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_float():
+        from numpy import real, shape, array, float
+        arr = array([4,5,6,2,1], float)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_float32():
+        from numpy import real, shape, array, float32
+        arr = array([4,5,6,2,1], float32)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_float64():
+        from numpy import real, shape, array, float64
+        arr = array([4,5,6,2,1], float64)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_complex64():
+        from numpy import real, shape, array, complex64
+        arr = array([4,5,6,2,1], complex64)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    def test_complex128():
+        from numpy import real, shape, array, complex128
+        arr = array([4,5,6,2,1], complex128)
+        a = real(arr)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    # gfortran complains about boolean to numeric convertions, given that boolean type in Fortran
+    # is not a numerical type, so Real function in Fortran doesn't accept a non-numerical type in the first argument
+    if (language == 'c'):
+        f_bl = epyccel(test_bool, language=language)
+        assert (f_bl() == test_bool())
+
+    f_integer = epyccel(test_int, language=language)
+    # int8 and int16 numpy data types not recognised by Pyccel.
+    #f_integer8 = epyccel(test_int8, language=language)
+    #f_integer16 = epyccel(test_int16, language=language)
```

Comment:
Why is this not a problem in the tests above? Do we accept `int8` arguments but not allow the creation of an array of type `int8`? That is strange ^^'

Reply:
If that is the case then could you open an issue? This is silly and they are easy to add